cmd/goatcounter/serve.go

```diff
@@ -164,18 +169,26 @@ func cmdServe(f zli.Flags, ready chan<- struct{}, stop chan struct{}) error {
 	var (
 		// TODO(depr): -port is for compat with <2.0
 		port         = f.Int(0, "public-port", "port").Pointer()
+		basePath     = f.String("/", "base-path").Pointer()
 		domainStatic = f.String("", "static").Pointer()
 	)
 	dbConnect, dbConn, dev, automigrate, listen, flagTLS, from, websocket, apiMax, err := flagsServe(f, &v)
 	if err != nil {
 		return err
 	}
 
-	return func(port int, domainStatic string) error {
+	return func(port int, basePath, domainStatic string) error {
 		if flagTLS == "" {
 			flagTLS = map[bool]string{true: "http", false: "acme,rdr"}[dev]
 		}
 
+		if !strings.HasPrefix(basePath, "/") {
+			return fmt.Errorf("invalid -base-path flag: %q: must start with a slash", basePath)
+		}
+		if i := len(basePath) - 1; basePath[i] == '/' {
+			basePath = basePath[:i]
+		}
```

Comment:
This can be a bit simpler with `basePath = strings.TrimRight(basePath, "/")`.

Actually I would probably just do `basePath = "/" + strings.Trim(base)` and remove that HasPrefix() check. If someone used `-base-path=stats` then it's pretty clear what they intended. No need to be pedantic about it.

Reply:
Good idea, changed to that.